src/coreComponents/linearAlgebra/interfaces/hypre/mgrStrategies/ThermalCompositionalMultiphaseReservoirFVM.hpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file ThermalCompositionalMultiphaseFVM.hpp
+ */
+
+#ifndef GEOS_LINEARALGEBRA_INTERFACES_HYPREMGRTHERMALCOMPOSITIONALMULTIPHASERESERVOIRFVM_HPP_
+#define GEOS_LINEARALGEBRA_INTERFACES_HYPREMGRTHERMALCOMPOSITIONALMULTIPHASERESERVOIRFVM_HPP_
+
+#include "linearAlgebra/interfaces/hypre/HypreMGR.hpp"
+
+namespace geos
+{
+
+namespace hypre
+{
+
+namespace mgr
+{
+
+/**
+ * @brief ThermalCompositionalMultiphaseReservoirFVM strategy.
+ *
+ * Labels description stored in point_marker_array
+ *               0 = pressure
+ *               1 = density
+ *             ... = densities
+ *   numLabels - 2 = last density
+ *   numLabels - 1 = temperature
+ *
+ * 2-level MGR reduction strategy
+ *   - 1st level: eliminate the reservoir density associated with the volume constraint
+ *   - 2nd level: eliminate the other reservoir densities
+ *   - The coarse grid (pressure and temperature system) is solved with BoomerAMG with numFunctions==2.
+ *
+ */
+class ThermalCompositionalMultiphaseReservoirFVM : public MGRStrategyBase< 3 >
+{
+public:
+  /**
+   * @brief Constructor.
+   * @param numComponentsPerField array with number of components for each field
+   */
+  explicit ThermalCompositionalMultiphaseReservoirFVM( arrayView1d< int const > const & numComponentsPerField )
```

Comment:
CC @castelletto1 and @victorapm and @jafranc

Reply:
@victorapm and @castelletto1 can you guys please confirm that the recipe  looks good?

Reply:
It looks good! @tjb-ltk can you just update the comments (well info is missing). We should have something like this plus temperature info https://github.com/GEOS-DEV/GEOS/blob/713046244c75a48eb198b30497c49a190433c88d/src/coreComponents/linearAlgebra/interfaces/hypre/mgrStrategies/CompositionalMultiphaseReservoirFVM.hpp#L34-L53